Source/AliveLibAO/Bullet.cpp

```diff
@@ -35,4 +43,202 @@ BaseGameObject* Bullet::VDestructor(signed int flags)
+                    }
+                }
+                if (!pShotObj || (distShot > distHit))
+                {
+                    if (field_20 <= FP_FromInteger(0))
+                    {
+                        auto pSpark = ao_new<Spark>();
+                        if (pSpark)
+                        {
+                            pSpark->ctor_477B70(hitX, hitY, field_2C_scale, 6u, -76, 76);
+                        }
+                    }
+                    else
+                    {
+                        auto pSpark = ao_new<Spark>();
+                        if (pSpark)
+                        {
+                            pSpark->ctor_477B70(hitX, hitY, field_2C_scale, 6u, 50, 205);
+                        }
+                    }
+                    New_Particles_419A80(hitX, hitY, field_2C_scale, 3, 0);
+                    if (Math_RandomRange_450F20(0, 100) >= 90)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    if (Math_RandomRange_450F20(0, 128) < 64)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    SFX_Play_43AD70(SoundEffect::Unknown_2, volume, 0);
+                    field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                    return;
+                }
+            }
+            else
+            {
+                if (!pShotObj)
+                {
+                    if ((signed __int16) Math_RandomRange_450F20(0, 100) >= 70)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    if (Math_RandomRange_450F20(0, 128) < 64)
+                    {
+                        SFX_Play_43AD70(SoundEffect::Unknown_1, volume, 0);
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    SFX_Play_43AD70(SoundEffect::Unknown_2, volume, 0);
+                    field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                    return;
+                }
+            }
+            if (pShotObj->VTakeDamage_401920(this))
+            {
+                SFX_Play_43AE60(SoundEffect::AirStream_28, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, 2000, 0);
+                SFX_Play_43AE60(SoundEffect::MeatBounce_43, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, Math_RandomRange_450F20(300, 700), 0);
+                SFX_Play_43AE60(SoundEffect::KillEffect_64, field_2C_scale != FP_FromDouble(0.5) ? 90 : 60, Math_RandomRange_450F20(900, 1400), 0);
+            }
+            field_6_flags.Set(BaseGameObject::eDead_Bit3);
+        }
+        case BulletType::Type_2:
+        {
+            FP v5;
+            FP v7;
```

Comment:
can v5 and v7 be named?

Reply:
They absolutely can. To what, though?

Reply:
they're only ever used as v7 - v5 twice, which is the rect.x pos, so v7 is always hero_xpos and v5 is elum speed or elum x offset?